src/util/rop_rad.jl

```diff
@@ -0,0 +1,276 @@
+                        end
+                    end
+
+                    Memento.info(_PM._LOGGER, "better order succeded")
+                    Memento.info(_PM._LOGGER, "new ENS: $(sum(values(new_ens)))")
+                    Memento.info(_PM._LOGGER, "old ENS: $(sum(values(old_ens)))")
+                else
+                    # insert old order, ENS was not improved
+                    for nwid in network_ids
+                        new_repair_ordering["$nwid"] = deepcopy(repair_ordering["$nwid"])
+                    end
+                    Memento.warn(_PM._LOGGER, "Failed to improve ENS")
+                    Memento.info(_PM._LOGGER, "new ENS: $(sum(values(new_ens)))")
+                    Memento.info(_PM._LOGGER, "old ENS: $(sum(values(old_ens)))")
+                end
+            else ## no primal solution to rop
+                new_ens = deepcopy(old_ens)
+
+                # insert old order
+                for nwid in network_ids
+                    new_repair_ordering["$nwid"] = deepcopy(repair_ordering["$nwid"])
+                end
+
+                Memento.info(_PM._LOGGER, "No Solution")
+                Memento.info(_PM._LOGGER, "Primal status: $(rad_solution["primal_status"])")
+                Memento.info(_PM._LOGGER, "Termination status: $(rad_solution["termination_status"])")
+            end
+
+            ## update running averages
+            if (sum(values(new_ens)) < sum(values(old_ens)))
+                push!(fail_to_improve, false)
+            else
+                push!(fail_to_improve, true)
+                average_fail_to_improve = average_fail_to_improve*(averaging_window-1)/averaging_window + 1/averaging_window
+            end
+            if rad_solution["termination_status"]!=_PM.TIME_LIMIT
+                push!(termination_time_limit, false)
+            else
+                push!(termination_time_limit, true)
+            end
+
+
+            push!(stats["solve_time"][iteration_counter], rad_solution["solve_time"])
+            push!(stats["termination_status"][iteration_counter], rad_solution["termination_status"])
+            push!(stats["primal_status"][iteration_counter], rad_solution["primal_status"])
+            push!(stats["ENS"][iteration_counter], sum(values(ens_dict)))
+            push!(stats["sub_ENS"][iteration_counter], sum(values(new_ens)))
+            push!(stats["improvement"][iteration_counter], sum(values(old_ens))-sum(values(new_ens)))
+            push!(stats["solver_time_limit"][iteration_counter], solver_time_limit)
+            push!(stats["partition_max"][iteration_counter], partition_max)
+            push!(stats["partition_size"][iteration_counter], repair_periods)
+            push!(stats["repair_count"][iteration_counter], count_repairable_components(r_data))
+        end
+
+        # Calc average and reset
+        average_fail_to_improve = Statistics.mean(fail_to_improve)
+        average_termination_time_limit = Statistics.mean(termination_time_limit)
```

Comment:
If the only function in Statistics that we need is `mean` I would define a version locally in this package or expand these two call with `sum(fail_to_improve)/length(fail_to_improve)`.